pytorch_lightning/trainer/connectors/debugging_connector.py

```diff
@@ -31,16 +31,32 @@ def on_init_start(
             overfit_batches,
             fast_dev_run
     ):
+        if not isinstance(fast_dev_run, (bool, int)):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a valid configuration.'
+                ' It should be either a bool or an int >= 0'
+            )
+
+        if isinstance(fast_dev_run, int) and (fast_dev_run < 0):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a'
+                ' valid configuration. It should be >= 0.'
+            )
 
         self.trainer.fast_dev_run = fast_dev_run
-        if self.trainer.fast_dev_run:
-            limit_train_batches = 1
-            limit_val_batches = 1
-            limit_test_batches = 1
+
+        if fast_dev_run is True:
+            fast_dev_run = 1
```

Comment:
You could avoid a line with :smile: 

```suggestion
        fast_dev_run = int(fast_dev_run)
```

Reply:
true, where are my python skills :smile: